File: AmtOrk.Server/Models/Credit.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AmtOrk.Server.Models
{
	public class Credit : RootModel
	{
		public DateOnly CreditDate { get; set; }

		[ForeignKey("CreditKingdom")]
		public virtual int CreditKingdomId { get; set; }

		[ForeignKey("CreditPark")]
		public virtual int CreditParkId { get; set; }

		[ForeignKey("CreditEvent")]
		public virtual int CreditEventId { get; set; }

		[ForeignKey("CreditClass")]
		public virtual int CreditClassId { get; set; }

		public int CreditCount { get; set; } = 1; // Credits default to 1, they have to be changed manually normally
	}
}

File: amtork.client/src/app/models/credit.ts
import { AmtgardEvent } from "./amtgardEvent";
import { Class } from "./class";
import { Kingdom } from "./kingdom";
import { Land } from "./land";


export interface Credit
{
    id:number;
    creditDate:Date;
    creditKingdomId:number;
    creditParkId:number;
    creditEventId:number | undefined;
    creditClassId:number;
    creditCount:number;
}

export const emptyCredit:Credit = {
    id: 0,
    creditDate: new Date(),
    creditKingdomId: 0,
    creditParkId: 0,
    creditEventId: undefined,
    creditClassId: 0,
    creditCount: 0
}